src/main/java/br/com/carrental/control/UserController.java

```diff
@@ -0,0 +1,48 @@
+package br.com.carrental.control;
+
+import br.com.carrental.model.User;
+import br.com.carrental.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class UserController {
+    @Autowired
+    private UserRepository repository;
+
+    @GetMapping("/users")
+    public List<User> getUsers(){
+        return repository.findAll();
+    }
+
+    @GetMapping("/users/{id}")
+    public User getUser(@PathVariable Long id){
+        Optional<User> user = repository.findById(id);
+
+        if(!user.isPresent()) throw new RuntimeException();
```

Comment:
Aqui se vc lançar uma exceção no seu get, o status code que retornará vai ser 500 (que é de erro no servidor). O que vc quer aqui não seria retornar um 404 (not found)?  Aí vc teria que mudar o retorno do seu método para ResponseEntity (como no post) e usar ele pra retornar o http status desejado.

Reply:
Dei uma modificada, adicionei o retorno de 404 caso não seja encontrado o user em questao pelo get e pelo delete também.